assets/src/edit-story/components/checklist/checklistContent/priorityChecks.js

```diff
@@ -18,32 +18,44 @@
  */
 import { __ } from '@web-stories-wp/i18n';
 import PropTypes from 'prop-types';
+import { useEffect } from 'react';
 /**
  * Internal dependencies
  */
-import { PANEL_STATES } from '../tablist';
-import { ISSUE_TYPES } from './constants';
-import PublisherLogoSize from './checks/publisherLogoSize';
-import StoryMissingExcerpt from './checks/storyMissingExerpt';
-import StoryMissingTitle from './checks/storyMissingTitle';
-import StoryPosterAspectRatio from './checks/storyPosterAspectRatio';
-import { StoryPosterAttached } from './checks/storyPosterAttached';
-import StoryPosterPortraitSize from './checks/storyPosterPortraitSize';
-import StoryTitleLength from './checks/storyTitleLength';
-import VideoElementMissingPoster from './checks/videoElementMissingPoster';
-import {
-  ChecklistCategoryProvider,
-  useCategoryCount,
-} from './checkCountContext';
-import { PanelText, StyledTablistPanel } from './styles';
+import { PANEL_STATES } from '../../tablist';
+import { ISSUE_TYPES, PANEL_VISIBILITY_BY_STATE } from '../constants';
+import PublisherLogoSize from '../checks/publisherLogoSize';
+import StoryMissingExcerpt from '../checks/storyMissingExerpt';
+import StoryMissingTitle from '../checks/storyMissingTitle';
+import StoryPosterAspectRatio from '../checks/storyPosterAspectRatio';
+import { StoryPosterAttached } from '../checks/storyPosterAttached';
+import StoryPosterPortraitSize from '../checks/storyPosterPortraitSize';
+import StoryTitleLength from '../checks/storyTitleLength';
+import VideoElementMissingPoster from '../checks/videoElementMissingPoster';
+import { ChecklistCategoryProvider, useCategoryCount } from '../countContext';
+import { PanelText, StyledTablistPanel } from '../styles';
+import { useCheckpoint } from '../checkpointContext';
 
 export function PriorityChecks({ isOpen, onClick, title }) {
   const count = useCategoryCount(ISSUE_TYPES.PRIORITY);
+  const { updateHighPriorityCount, checkpoint } = useCheckpoint(
+    ({ actions: { updateHighPriorityCount }, state: { checkpoint } }) => ({
+      checkpoint,
+      updateHighPriorityCount,
+    })
+  );
+  useEffect(() => {
+    updateHighPriorityCount(count);
+  }, [updateHighPriorityCount, count]);
```

Comment:
Do we need this effect here? The other two check components (design, accessibility) don't have this kind of effect so that feels like we shouldn't need something like this.

To me it feels like this should be calculated by the checklist provider, right?

Reply:
yeah, good catch. I'll remove it!

Reply:
But what I'm wondering is do we need to life the `useCategoryCount` context up since this is updating state higher up in the tree 🤔

Reply:
I was trying to keep the count context tight to the checklist, since it's just this 1 count that needs to be seen in 1 place super high up in the tree. I tried to outline it in the documentation I added?  Super open to restructuring, we've just worked so hard to tighten this all up that I was hesitant to really lift much. 

```
- CheckpointProvider: Wraps around Layout so that the publish  button also has access to the current checkpoint. If requirements are met then a dialog appears prompting the  user to review the checklist before publishing.
- ChecklistProvider: Wraps around the carousel so that the toggle buttons in the secondary menu know when the checklist is open and the other popup menus there (help center) can respect it. 
- ChecklistCountProvider, ChecklistCategoryProvider: Wrap around various parts of the checklist itself to provide counts of issues.
```